lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
@kjdelisle there is another method called `update/updateAll`, I think you are actually looking for that: https://github.com/strongloop/loopback-connector-cloudant/blob/347a75289abaeb9040f64e43195144d68e8c711d/lib/cloudant.js#L770